npm v7: peer mocha@">=1.18 <7" from leche@2.3.0

when running `npm install` with npm v7.
```bash
➜  espree git:(main) npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: espree@7.3.1
npm ERR! Found: mocha@8.3.2
npm ERR! node_modules/mocha
npm ERR!   dev mocha@"^8.3.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mocha@">=1.18 <7" from leche@2.3.0
npm ERR! node_modules/leche
npm ERR!   dev leche@"^2.3.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/weiran/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/weiran/.npm/_logs/2021-04-13T03_23_03_200Z-debug.log
```

we have to remove leche(like in the eslint repo) - it has been unmaintained.

--- a/package.json
+++ b/package.json
@@ -48,7 +48,6 @@
     "esprima": "latest",
     "esprima-fb": "^8001.2001.0-dev-harmony-fb",
     "json-diff": "^0.5.4",
-    "leche": "^2.3.0",
     "mocha": "^8.3.1",
     "npm-run-all": "^4.1.5",
     "nyc": "^14.1.1",

--- a/tests/lib/ecma-features.js
+++ b/tests/lib/ecma-features.js
@@ -57,30 +57,31 @@ describe("ecmaFeatures", () => {
         };
     });
     testFiles.forEach(filename => {
+        describe(filename, () => {
 
-        // Uncomment and fill in filename to focus on a single file
-        // var filename = "jsx/invalid-matching-placeholder-in-closing-tag";
-        const feature = path.dirname(filename),
-            isPermissive = !shouldThrowInTestsWhenEnabled(feature),
-            code = shelljs.cat(`${path.resolve(FIXTURES_DIR, filename)}.src.js`);
+            // Uncomment and fill in filename to focus on a single file
+            // var filename = "jsx/invalid-matching-placeholder-in-closing-tag";
+            const feature = path.dirname(filename),
+                isPermissive = !shouldThrowInTestsWhenEnabled(feature),
+                code = shelljs.cat(`${path.resolve(FIXTURES_DIR, filename)}.src.js`);
 
-        it(`should parse correctly when ${feature} is ${isPermissive}`, async () => {
-            config.ecmaFeatures[feature] = isPermissive;
+            it(`should parse correctly when ${feature} is ${isPermissive}`, async () => {
+                config.ecmaFeatures[feature] = isPermissive;
 
-            // eslint-disable-next-line node/no-unsupported-features/es-syntax
-            const expected = await import(`${pathToFileURL(path.resolve(__dirname, "../../", FIXTURES_DIR, filename)).href}.result.js`);
+                // eslint-disable-next-line node/no-unsupported-features/es-syntax
+                const expected = await import(`${pathToFileURL(path.resolve(__dirname, "../../", FIXTURES_DIR, filename)).href}.result.js`);
 
-            tester.assertMatches(code, config, expected.default);
-        });
-
-        it(`should throw an error when ${feature} is ${!isPermissive}`, () => {
-            config.ecmaFeatures[feature] = !isPermissive;
-
-            assert.throws(() => {
-                espree.parse(code, config);
+                tester.assertMatches(code, config, expected.default);
             });
 
-        });
+            it(`should throw an error when ${feature} is ${!isPermissive}`, () => {
+                config.ecmaFeatures[feature] = !isPermissive;
 
+                assert.throws(() => {
+                    espree.parse(code, config);
+                });
+
+            });
+        });
     });
 });

--- a/tests/lib/ecma-features.js
+++ b/tests/lib/ecma-features.js
@@ -8,7 +8,6 @@
 //------------------------------------------------------------------------------
 
 import assert from "assert";
-import leche from "leche";
 import path from "path";
 import * as espree from "../../espree.js";
 import shelljs from "shelljs";
@@ -57,8 +56,7 @@ describe("ecmaFeatures", () => {
             ecmaFeatures: {}
         };
     });
-
-    leche.withData(testFiles, filename => {
+    testFiles.forEach(filename => {
 
         // Uncomment and fill in filename to focus on a single file
         // var filename = "jsx/invalid-matching-placeholder-in-closing-tag";
@@ -83,5 +81,6 @@ describe("ecmaFeatures", () => {
             });
 
         });
+
     });
 });

--- a/tests/lib/ecma-version.js
+++ b/tests/lib/ecma-version.js
@@ -59,21 +59,24 @@ describe("ecmaVersion", () => {
 
     describe("Scripts", () => {
         scriptOnlyTestFiles.forEach(filename => {
-            const version = filename.slice(1, filename.indexOf("/", 1));
+            describe(filename, () => {
+                const version = filename.slice(1, filename.indexOf("/", 1));
 
-            // Uncomment and fill in filename to focus on a single file
-            // var filename = "newTarget/simple-new-target";
-            const code = shelljs.cat(`${FIXTURES_DIR}/${filename}.src.js`);
+                // Uncomment and fill in filename to focus on a single file
+                // var filename = "newTarget/simple-new-target";
+                const code = shelljs.cat(`${FIXTURES_DIR}/${filename}.src.js`);
 
-            it("should parse correctly when sourceType is script", async () => {
-                config.ecmaVersion = Number(version);
+                it("should parse correctly when sourceType is script", async () => {
+                    config.ecmaVersion = Number(version);
 
-                const absolutePath = path.resolve(__dirname, FIXTURES_DIR, filename.slice(1));
-                // eslint-disable-next-line node/no-unsupported-features/es-syntax
-                const expected = await import(`${pathToFileURL(absolutePath).href}.result.js`);
+                    const absolutePath = path.resolve(__dirname, FIXTURES_DIR, filename.slice(1));
+                    // eslint-disable-next-line node/no-unsupported-features/es-syntax
+                    const expected = await import(`${pathToFileURL(absolutePath).href}.result.js`);
 
-                tester.assertMatches(code, config, expected.default);
+                    tester.assertMatches(code, config, expected.default);
+                });
             });
+
         });
 
     });
@@ -81,36 +84,39 @@ describe("ecmaVersion", () => {
 
     describe("Modules", () => {
         moduleTestFiles.forEach(filename => {
-            const version = filename.slice(1, filename.indexOf("/", 1));
-            const code = shelljs.cat(`${FIXTURES_DIR}/${filename}.src.js`);
+            describe(filename, () => {
+                const version = filename.slice(1, filename.indexOf("/", 1));
+                const code = shelljs.cat(`${FIXTURES_DIR}/${filename}.src.js`);
 
-            it("should parse correctly when sourceType is module", async () => {
-                const absolutePath = path.resolve(__dirname, FIXTURES_DIR, filename.slice(1));
+                it("should parse correctly when sourceType is module", async () => {
+                    const absolutePath = path.resolve(__dirname, FIXTURES_DIR, filename.slice(1));
 
-                let expected;
+                    let expected;
 
-                try {
-                    // eslint-disable-next-line node/no-unsupported-features/es-syntax
-                    expected = await import(`${pathToFileURL(absolutePath).href}.module-result.js`);
-                } catch {
-                    // eslint-disable-next-line node/no-unsupported-features/es-syntax
-                    expected = await import(`${pathToFileURL(absolutePath).href}.result.js`);
-                }
+                    try {
+                        // eslint-disable-next-line node/no-unsupported-features/es-syntax
+                        expected = await import(`${pathToFileURL(absolutePath).href}.module-result.js`);
+                    } catch {
+                        // eslint-disable-next-line node/no-unsupported-features/es-syntax
+                        expected = await import(`${pathToFileURL(absolutePath).href}.result.js`);
+                    }
 
-                if (expected) {
-                    expected = expected.default;
-                }
+                    if (expected) {
+                        expected = expected.default;
+                    }
 
-                config.ecmaVersion = Number(version);
-                config.sourceType = "module";
+                    config.ecmaVersion = Number(version);
+                    config.sourceType = "module";
 
-                // set sourceType of program node to module
-                if (expected.type === "Program") {
-                    expected.sourceType = "module";
-                }
+                    // set sourceType of program node to module
+                    if (expected.type === "Program") {
+                        expected.sourceType = "module";
+                    }
 
-                tester.assertMatches(code, config, expected);
+                    tester.assertMatches(code, config, expected);
+                });
             });
+
         });
     });
 

--- a/tests/lib/ecma-version.js
+++ b/tests/lib/ecma-version.js
@@ -8,7 +8,6 @@
 //------------------------------------------------------------------------------
 
 import fs from "fs";
-import leche from "leche";
 import path from "path";
 import shelljs from "shelljs";
 import tester from "./tester.js";
@@ -59,8 +58,7 @@ describe("ecmaVersion", () => {
     });
 
     describe("Scripts", () => {
-
-        leche.withData(scriptOnlyTestFiles, filename => {
+        scriptOnlyTestFiles.forEach(filename => {
             const version = filename.slice(1, filename.indexOf("/", 1));
 
             // Uncomment and fill in filename to focus on a single file
@@ -76,15 +74,13 @@ describe("ecmaVersion", () => {
 
                 tester.assertMatches(code, config, expected.default);
             });
-
         });
 
     });
 
 
     describe("Modules", () => {
-
-        leche.withData(moduleTestFiles, filename => {
+        moduleTestFiles.forEach(filename => {
             const version = filename.slice(1, filename.indexOf("/", 1));
             const code = shelljs.cat(`${FIXTURES_DIR}/${filename}.src.js`);
 
@@ -115,7 +111,6 @@ describe("ecmaVersion", () => {
 
                 tester.assertMatches(code, config, expected);
             });
-
         });
     });
 

--- a/tests/lib/libraries.js
+++ b/tests/lib/libraries.js
@@ -30,20 +30,23 @@ const testFiles = shelljs.find(`${__dirname}/../fixtures/libraries`).filter(file
 
 describe("Libraries", () => {
     testFiles.forEach(filename => {
+        describe(filename, () => {
 
-        // var filename = "angular-1.2.5.js";
+            // var filename = "angular-1.2.5.js";
 
-        it("should produce correct AST when parsed", () => {
-            const output = shelljs.cat(`${filename}.result.json`);
-            const input = shelljs.cat(filename);
+            it("should produce correct AST when parsed", () => {
+                const output = shelljs.cat(`${filename}.result.json`);
+                const input = shelljs.cat(filename);
 
-            const result = JSON.stringify(tester.getRaw(espree.parse(input, {
-                loc: true,
-                range: true,
-                tokens: true
-            })));
+                const result = JSON.stringify(tester.getRaw(espree.parse(input, {
+                    loc: true,
+                    range: true,
+                    tokens: true
+                })));
 
-            assert.strictEqual(result, output);
+                assert.strictEqual(result, output);
+            });
         });
+
     });
 });

--- a/tests/lib/libraries.js
+++ b/tests/lib/libraries.js
@@ -7,7 +7,6 @@
 // Requirements
 //------------------------------------------------------------------------------
 
-import leche from "leche";
 import path from "path";
 import shelljs from "shelljs";
 import tester from "./tester.js";
@@ -30,8 +29,7 @@ const testFiles = shelljs.find(`${__dirname}/../fixtures/libraries`).filter(file
 //------------------------------------------------------------------------------
 
 describe("Libraries", () => {
-
-    leche.withData(testFiles, filename => {
+    testFiles.forEach(filename => {
 
         // var filename = "angular-1.2.5.js";
 
@@ -48,6 +46,4 @@ describe("Libraries", () => {
             assert.strictEqual(result, output);
         });
     });
-
-
 });

